Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp

```diff
@@ -2230,9 +2233,20 @@ void InstrumentDefinitionParser::setLogfile(
       continue;
     }
 
+    DateAndTime validityDate;
+
+    if (requestedDate.empty()) {
+      validityDate = DateAndTime::getCurrentTime();
+    } else {
+      validityDate.setFromISO8601(requestedDate);
```

Comment:
Here you could log an error if requestedDate doesn't appear to be in ISO format.